src/main/java/org/opensearch/knn/index/codec/VectorSerializerFactory.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.codec;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.io.ByteArrayInputStream;
+import java.io.ObjectStreamConstants;
+import java.util.Arrays;
+import java.util.Map;
+
+import static org.opensearch.knn.index.codec.VectorSerializerFactory.SerializationMode.ARRAY;
```

Comment:
Why do we need to import these if they are in this file? Could SerializationMode be its own file if it is public?

Reply:
I use this notation as it's shorter comparing to SerializationMode.ARRAY form, because the enum is referenced inside the Factory class multiple times. 
I usually try to keep enum inside the class if it's just the DTO, comparing to other cases when enum class may have additional method with it's own logic - for such cases I agree its cleaner to have enum as a separate class.  
Just checked k-NN plugin codebase, seems de-facto standard is to have it a separate class, let me change it in next commit

Reply:
made the change, resolving